tests/smoke-tests/test-staking-rewards.ts

```diff
@@ -288,15 +301,52 @@ async function assertRewardsAt(api: ApiPromise, nowBlockNumber: number) {
     ).to.be.empty;
   }
 
+  // Perbill arithmetic can deviate at most ±1 per operation so we use the number of collators
+  // to compute the max deviation per billion
+  const maxDifference = awardedCollatorCount;
+
+  // assert rewarded amounts match (with loss due to Perbill arithmetic)
+  const estimatedCommissionRewardedLoss = new Perbill(BN_BILLION.sub(totalCollatorShare)).of(
+    totalCollatorCommissionReward
+  );
+  const actualCommissionRewardedLoss = totalCollatorCommissionReward.sub(
+    totalCollatorCommissionRewarded
+  );
+  expect(
+    estimatedCommissionRewardedLoss.sub(actualCommissionRewardedLoss).abs().toNumber(),
+    `Percentage share loss was above ${maxDifference} parts per billion, \
+  estimated ${estimatedCommissionRewardedLoss.toString()}, \
+  actual ${actualCommissionRewardedLoss.toString()}`
+  ).to.be.lessThanOrEqual(maxDifference);
+
+  // we add the two estimated losses, since the totalBondReward is always split between N collators,
+  // which then split the reward again between the all the delegators
+  const estimatedBondRewardedLoss = new Perbill(BN_BILLION.sub(totalCollatorShare))
+    .of(totalBondReward)
+    .add(totalBondRewardedLoss);
+  const actualBondRewardedLoss = totalBondReward.sub(totalBondRewarded);
+  expect(
+    estimatedBondRewardedLoss.sub(actualBondRewardedLoss).abs().toNumber(),
+    `Percentage share loss was above ${maxDifference} parts per billion, \
+  estimated ${estimatedBondRewardedLoss.toString()}, actual ${actualBondRewardedLoss.toString()}`
+  ).to.be.lessThanOrEqual(maxDifference);
```

Comment:
Shouldnt the reward loss bound be higher? maxDIfference is total collator, but shouldnt this be totalCollators + totalDelegations?

Reply:
We check this already in the inner loop and return the real difference, so the losses shouldn't accumulate

Reply:
Oh I see, perfect